flink-python/pyflink/table/table_environment.py

```diff
@@ -1770,11 +1771,19 @@ def from_data_stream(self, data_stream: DataStream, fields: List[str] = None) ->
                        of the Table
         :return: The converted Table.
         """
-        if fields is not None:
-            j_table = self._j_tenv.fromDataStream(data_stream._j_data_stream, fields)
-        else:
+        j_table = None
+        if len(fields) == 0:
             j_table = self._j_tenv.fromDataStream(data_stream._j_data_stream)
-        return Table(j_table=j_table, t_env=self._j_tenv)
+        elif len(fields) == 1 and isinstance(fields[0], str):
+            j_table = self._j_tenv.fromDataStream(data_stream._j_data_stream, fields[0])
```

Comment:
What about adding a warn to tell the user that this method has been deprecated

Reply:
@HuangXingBo Yes, I would like to add warn log to tell users with this deprecated method.